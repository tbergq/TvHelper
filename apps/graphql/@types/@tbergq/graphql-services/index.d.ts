declare module '@tbergq/graphql-services';
