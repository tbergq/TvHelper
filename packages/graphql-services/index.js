// @flow

export { default as fetch } from './src/fetch';

// graphql shared types
export { default as LoginType } from './src/shared/types/output/LoginType';
export { default as RangeDelete } from './src/shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './src/shared/resolvers/LoginResolver';

export type { LoggedInUser } from './src/shared/resolvers/LoginResolver';
export type { Apps } from './src/shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './src/connectionFromArray';
export type { ConnectionArguments, Connection } from './src/connectionFromArray';
export type { GraphQLConnectionDefinitions } from './src/types';
