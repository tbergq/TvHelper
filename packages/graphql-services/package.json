{
  "name": "@tbergq/graphql-services",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "@adeira/fetch": "^1.0.0",
    "@kiwicom/graphql-global-id": "^0.13.0",
    "dataloader": "^1.4.0",
    "dotenv": "^8.2.0",
    "graphql": "^14.5.8",
    "graphql-relay": "^0.6.0",
    "jsonwebtoken": "^8.5.1",
    "password-hash": "^1.2.2"
  }
}
