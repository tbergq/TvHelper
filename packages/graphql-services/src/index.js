// @flow

export { default as fetch } from './fetch';

// graphql shared types
export { default as LoginType } from './shared/types/output/LoginType';
export { default as RangeDelete } from './shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './shared/resolvers/LoginResolver';

export type { LoggedInUser } from './shared/resolvers/LoginResolver';
export type { Apps } from './shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './connectionFromArray';
export type { ConnectionArguments, Connection } from './connectionFromArray';
export type { GraphQLConnectionDefinitions } from './types';
