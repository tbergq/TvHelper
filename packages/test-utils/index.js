// @flow

export { default as withMongoDb } from './src/useMongoDb';
