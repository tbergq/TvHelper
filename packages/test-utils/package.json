{
  "name": "@tbergq/test-utils",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "mongodb-memory-server": "^5.2.0",
    "mongoose": "^5.6.9"
  }
}
