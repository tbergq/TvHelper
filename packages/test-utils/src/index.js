// @flow

export { screen, fireEvent, act, waitFor } from '@testing-library/react';
export { default as render } from './render';
export { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
