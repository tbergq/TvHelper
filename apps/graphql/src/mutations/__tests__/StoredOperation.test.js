// @flow

import { generateTestsFromFixtures } from '@kiwicom/test-utils';
import { generateExecuteTestQuery } from '@tbergq/graphql-services';

import createdStoredOperations from '../StoredOperation';

const executeTestQuery = generateExecuteTestQuery(
  null,
  {
    createdStoredOperations,
  },
  {},
);

describe('mutations', () => {
  generateTestsFromFixtures(`${__dirname}/__fixtures__`, input => executeTestQuery(input));
});
