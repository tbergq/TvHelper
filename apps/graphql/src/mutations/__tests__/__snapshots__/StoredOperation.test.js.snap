// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations matches expected output: storedOperation.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation {
  createdStoredOperations(storedOperations: [{operationId: "123", text:"lol"},{operationId: "321", text:"lol2"}]) {
    createdOperations {
      operationId
      text
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "createdStoredOperations": {
      "createdOperations": [
        {
          "operationId": "123",
          "text": "lol"
        },
        {
          "operationId": "321",
          "text": "lol2"
        }
      ]
    }
  }
}
`;
